### PR TITLE
refactor ContextExt

### PR DESCRIPTION
**Pull request by Lonexera**

### CODE REVIEW

File: app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        var bytesRead: Int
+        while (input.read(buffer).also { bytesRead = it } != -1) {
+            output.write(buffer, 0, bytesRead)
+        }
+    }
+
+    sealed class SavingUriOutput {
+        data class Success(val uri: Uri) : SavingUriOutput()
+        object Failure : SavingUriOutput()
+    }
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt:51]: I think you can remove here this sealed class and just process this with try-catch in viewmodel

File: app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.example.plantsapp.presentation.ui.plantcreation
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt:1]: Also 2 points to mention: 
1. I think this is more like a data class - so put it into data layer
2. You can add interface to domain and use this interface in presentation
Also add package for usecase in domain and data

File: app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import java.io.InputStream
+import javax.inject.Inject
+
+class SaveUriInStorageUseCase @Inject constructor(
+    @ApplicationContext private val context: Context
+) {
+
+    fun saveImage(uri: Uri): SavingUriOutput {
+        val file = try {
+            val newFile = createImageFile(context) ?: throw IOException()
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt:20]: I do not like the idea of throwing IOException by ourselves 

Questions:
- why createImageFile can return null? Maybe there some other way do it?
- get into documentation for File.createTempFile and maybe it always return non nullable? and issue is with java?

File: app/src/main/java/com/example/plantsapp/presentation/ui/utils/ContextExt.kt [hunk 4]
```diff
@@ -50,7 +32,7 @@ private fun Context.getFileUri(file: File): Uri {
 }
 
 @Throws(IOException::class)
-private fun createImageFile(context: Context): File? {
+fun createImageFile(context: Context): File? {
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/utils/ContextExt.kt:35]: Leave here //TODO Maybe move to data package

File: app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/PlantCreationViewModel.kt [hunk 5]
```diff
@@ -76,10 +77,22 @@ class PlantCreationViewModel @Inject constructor(
...
     }
 
-    fun onImageSelected(uri: Uri) {
+    fun onImageCaptured(uri: Uri) {
         _selectedPicture.value = uri
     }
 
+    fun onImageSelected(uri: Uri) {
+        val savingResult = saveUriInStorageUseCase.saveImage(uri)
+        when (savingResult) {
```
Comment (Vladuken) [inline app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/PlantCreationViewModel.kt:86]: change to try-catch here

**Review by Vladuken — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt (hunk 3 above)
Comment (Lonexera) [reply app/src/main/java/com/example/plantsapp/presentation/ui/plantcreation/SaveUriInStorageUseCase.kt:20]: okay, yeah, File.createTempFile() never returns null. it just throws an exception if something went wrong
so, yes, we can return non-null from createImageFile() and remove this null check

**Pull request merged**